File: android/app/src/main/java/com/coinswallet/coins/FilecoinCoin.java
// FilecoinCoin.java
package com.coinswallet.coins;

import com.coinswallet.CoinFactory;

import wallet.core.jni.CoinType;
import wallet.core.jni.PrivateKey;

public class FilecoinCoin extends CoinFactory.Coin {
    private final String address = "";
    private final String privateKey = "";

    private final String defaultHDPath = "m/44'/461'/0'/0/0";
    private final String testDefaultHDPath = "m/44'/1'/0'/0/0";

    public FilecoinCoin(String mnemonic) {
        super(mnemonic);
    }

    @Override
    public String getNewAddress(Boolean isTestNet) {
        String path = isTestNet ? testDefaultHDPath : defaultHDPath;
        String derivedAddress = CoinType.FILECOIN.deriveAddress(wallet.getKey(CoinType.FILECOIN, path));
        if (isTestNet) {
            return "t" + derivedAddress.substring(1);
        }
        return derivedAddress;
    }

    @Override
    public String getPrivateKey(Boolean isTestNet) {
        String path = isTestNet ? testDefaultHDPath : defaultHDPath;
        return CoinFactory.Coin.toHex(wallet.getKey(CoinType.FILECOIN, path).data());
    }

    @Override
    public String signTransaction(String rawData) {
        return null;
    }
}

File: ios/FilecoinCoin.swift
//
//  FileCoin.swift
//  coinswallet
//
//  Created by Tal Grynbaum on 30/06/2023.
//

import Foundation
import WalletCore

class FilecoinCoin: CoinFactory.Coin {
  private var address: String = ""
  private var privateKey: String = ""

  private let defaultHDPath = "m/44\'/461\'/0\'/0/0"
  private let testDefaultHDPath = "m/44\'/1\'/0\'/0/0"

  override init(mnemonic: String) {
    super.init(mnemonic: mnemonic)
  }

  override func getNewAddress(isTestNet: Bool) -> String {
    let path = isTestNet ? testDefaultHDPath : defaultHDPath
    let privateKey = wallet.getKey(coin: .filecoin, derivationPath: path)
    let address = CoinType.filecoin.deriveAddress(privateKey: privateKey)
    if (isTestNet) {
      var modified = address
      modified.replaceSubrange(modified.startIndex...modified.startIndex, with: "t")
      return modified
    }
    return address
  }

  override func getPrivateKey(isTestNet: Bool) -> String {
    let path = isTestNet ? testDefaultHDPath : defaultHDPath
    let privateKey = wallet.getKey(coin: .filecoin, derivationPath: path)
    return privateKey.data.hexString
  }

  override func signTransaction(rawData: String) -> String {
    // Implement the logic to sign an Ethereum transaction using the provided raw data
    return ""
  }
}
